### ISSUE TITLE
Wallet API improvements

### ISSUE THREAD

**Issue by iscander01** [BeamMW/beam#528]
1. Need to add the following parameters to wallet API methods:
- tx_list - date of transaction;
- get_utxo - status

2. Add API methods for:
- get addresses list
- delete address
- edit address
- get contacts list
- remove contact

**Comment by iscander01** [comment]
@nesbox, please also add error message to tx_list results, and if it possible type of transaction (receiving\sending)

**Comment by nesbox** [comment]
- [x] tx_list - date of transaction;
- [x] 
- [x] 

get_utxo - status
Add API methods for:
get addresses list
delete address
edit address
get contacts list
remove contact

**Comment by nesbox** [comment]
- [x] added date of a transaction to `tx_list` api.
- [x] added status to `get_utxo` api.
- [x] added `addr_list(own)` api method.
- [x] added `delete_address(id)` method.
- [x] added `edit_address(id)` method.
- [x] get contacts list - just call `addr_list(false)` to get contacts list.
- [x] use `delete_address(id)` to delete contact.

Also, Wallet API Wiki should be updated after release.


## addr_list
Get all the addresses in the wallet. Use `own=true` to get all the own addresses

`-->`
```json
{
	"jsonrpc":"2.0", 
	"id": 8,
	"method":"addr_list",
	"params":
	{
		"own" : true
	}
}
```
`<--`
```json
{
	"id": 8,
	"jsonrpc": "2.0",
	"result":
	[
		{
			"address":
			"29510b33fac0cb20695fd3b836d835451e600c4224d8fb335dc1a68271deb9b6b5b",
			"category": "",
			"create_time": 1553174321,
			"duration": 1520,
			"expired": true,
			"label": "",
			"own": true 
		}
	] 
}
```

## delete_addr
Delete specific address from the wallet.

`-->`
```json
{
	"jsonrpc":"2.0", 
	"id": 8,
	"method":"delete_addr",
	"params":
	{
		"address" : "29510b33fac0cb20695fd3b836d835451e600c4224d8fb335dc1a68271deb9b6b5b"
	}
}
```
`<--`
```json
{
	"id": 8,
	"jsonrpc": "2.0",
	"result": "done"
}
```

## edit_addr
Edit specific address in the wallet, you can change the `label` or make it `expired/eternal/active` or do both things at the same time.

`-->`
```json
{
	"jsonrpc":"2.0", 
	"id": 8,
	"method":"delete_addr",
	"params":
	{
		"label" : "John Smith",
		"action" : "expired"
	}
}
```
`<--`
```json
{
	"id": 8,
	"jsonrpc": "2.0",
	"result": "done"
}
```

**Comment by nesbox** [comment]
> @nesbox, please also add error message to tx_list results, and if it possible type of transaction (receiving\sending)

Error message already added, pls look at `failure_reason` field.
Also, I'll add type of TX soon...

**Comment by nesbox** [comment]
Added `income` field to the `tx_status` api, which indicates that TX has receiving status

**Comment by Sergei-Beam** [comment]
Checking
- [x] added date of a transaction to `tx_list` api.
- [x] added status to `get_utxo` api.
- [x] added `addr_list(own)` api method.
- [x] added `delete_address(id)` method.
- [x] added `edit_address(id)` method.
- [x] get contacts list - just call `addr_list(false)` to get contacts list.
- [x] use `delete_address(id)` to delete contact.

**Comment by Sergei-Beam** [comment]
@nesbox  for new method 'edit_address' please change:
- label to comment (since cli wallet haven't label command)
- action to expiration: 24h/never/expired ( it'll look more clear and understandable)

**Comment by nesbox** [comment]
> @nesbox for new method `edit_address` please change:
> 
> * `label` to `comment` (since cli wallet haven't `label` command)
> * `action` to `expiration`: `24h/never/expired` ( it'll look more clear and understandable)

done here 098135b bdb0d50

**Comment by nesbox** [comment]
fixed empty comment field here 6ec933c

**Comment by Sergei-Beam** [comment]
checked
